greeter/README.md

```diff
@@ -139,67 +169,166 @@ A Controller is a class that implements operations defined by an application’s
    ```
 
-   Then add these lines as a mock greeting service to the controller
+2. In the generated GreetingController in `src/controllers/greeting.controller.ts`, modify the constructor:
 
    ```ts
-   // Will be replaced by the greeting service from extension later
-   export class SimpleGreetingService extends GreetingService {
-     async greet(language: string, name: string): Promise<string> {
-       return `${name} in language ${language}`;
-     }
+    private greetingService: SimpleGreetingService;
+
+    constructor() {
+      this.greetingService = new SimpleGreetingService();
+    }
+   ```
+
+3. Add the endpoint `/greet/{name}` which calls `SimpleGreetingService`. It only handles the English language and the Chinese language. If any other language is specified, it falls back to English.
+
+   ```ts
+    @get('/greet/{name}', {
+      responses: {
+      '200': {
+          description: '',
+          content: {
+          'application/json': {
+              schema: {
+              type: 'object',
+              properties: {
+                  timestamp: 'string',
+                  language: 'string',
+                  message: 'string',
+              },
+              },
+          },
+          },
+      },
+      },
+      })
+    async greet(
+      @param.path.string('name') name: string,
+      @param.header.string('Accept-Language') lan: string,
+    ): Promise<Message> {
+      const language: string = lan;
+      const greeting = await this.greetingService.greet(language, name);
+      return {
+      timestamp: new Date(),
+      language,
+      greeting,
+      };
    }
    ```
-   -- this is actually a bad design. As we showed in the diagram, we want to decouple our app as much as possible. We will show how to separate the greeting service from the controller later. 
 
-2. In the generated GreetingController in `src/controllers/greeting.controller.ts`, modify the constructor:
+4. Add the necessary imports
 
-   The `@inject` decorator here is a usage of Context and IoC, it binds components to your app. And a binding connects its value to a unique key as the address to access the entry in a context. Such injections/bindings are useful in real world application design. In our case, instead of having the controller handle all logic, Context and IoC decouple the execution and gives the application more modularity.
+   ```ts
+    import {SimpleGreetingService} from '../simple.greeting.service';
+    import {param, get} from '@loopback/rest';
+    import {Message} from '../types';
+   ```
+
+
+### Step 5: Try it Out!
+
+Start the app.
+
+```sh
+$ npm start
+Server is running at http://[::1]:3000
+Try http://[::1]:3000/ping
+```
+
+Go to the [API Explorer](http://localhost:3000/explorer), you should see the `GreetingController` section as shown in the screen shot below. Then click "Try it out".
+
+![Test endpoint in APIExplorer](images/test-greeting-1.png)
+
+#### Test 1: Set the Language to English `en`
```

Comment:
I think to test 3 languages in both part 1 and part 2 are kind of redundant. Maybe test `fr` on part 2 is enough to show the idea of these 2 parts.

(Feel free to ignore if you will simply it during presentation and just add details to the instruction)